src/ai/backend/manager/models/gql_models/agent.py

```diff
@@ -722,11 +722,14 @@ async def batch_load(
         graph_ctx: GraphQueryContext,
         agent_ids: Sequence[AgentId],
         *,
-        domain_name: str | None,
+        access_key: Optional[str] = None,
+        domain_name: Optional[str] = None,
         raw_status: Optional[str] = None,
         scaling_group: Optional[str] = None,
-        access_key: str,
-    ) -> Sequence[Agent | None]:
+    ) -> Sequence[Self | None]:
+        if access_key is None:
+            raise ValueError("access_key is required for the AgentSummary.batch_load().")
+
```

Comment:
In the previous behavior, access_key was already optional, right?

Reply:
No, `access_key` is required. However, due to the below type issue, it has been changed to Optional.
(And if `access_key` is None at runtime, raise the ValueError.)

See https://github.com/lablup/backend.ai/pull/3045#discussion_r1832175374.

Reply:
Ah... `get_loader_by_func` takes kwargs and generalizes it even if the function signatures are different.
I understand.

I think that generalizing `get_loader_by_func` to receive different kwargs for each `batch_load_func` might make debugging more difficult.